### PR TITLE
[curves tracing] expose settings for tracing curves in the UI

### PR DESCRIPTION
**Pull request by olivierdalang**
## Description

This exposes the settings used when tracing curves in the main settings dialog.

These setting did already exist (introduced by https://github.com/qgis/QGIS/pull/37826) but so far were not exposed in the UI, as they are rather technical. Yet, experience shows it's still relatively common that these settings must be tweaked by users to get good results.

If possible, we'd like to backport this to 3.16.

image

### CODE REVIEW

File: src/ui/qgsoptionsbase.ui [hunk 1]
```diff
@@ -4888,6 +4888,46 @@ The bigger the number, the faster zooming with the mouse wheel will be.</string>
...
+                   <widget class="QLabel" name="label_70">
+                    <property name="text">
+                     <string>Distance tolerance when tracing curves</string>
+                    </property>
+                   </widget>
+                  </item>
+                  <item row="1" column="2">
+                   <widget class="QgsDoubleSpinBox" name="mTracingCustomAngleToleranceSpinBox">
+                    <property name="toolTip">
+                     <string>Parameter used by the convert to curves algorithm when tracing curves.</string>
```
Comment (m-kuhn) [inline src/ui/qgsoptionsbase.ui:4908]:
```suggestion
                     <string>Parameter used by the convert to curves algorithm when tracing curves.</string>
```

Could this be a tiny bit more explanatory? I don't know if you have a good wording for *how* it works, but already knowing what happens if it's lower or greater (more tolerance vs. stricter lower chance of unintentional detection of curves) would be nice.

Comment (m-kuhn) [reply src/ui/qgsoptionsbase.ui:4908]: Other things to potentially mention:

 - rounding / floating point
 - depends on significant digits in the coordinate values

Comment (olivierdalang) [reply src/ui/qgsoptionsbase.ui:4908]: Yes it's worth explaining, but wonder if the tooltip is the best place to do so ? Due to the complexity of the setting, it's hard to have anything useful in not too many lines...

Maybe we could improve the help text of the convert to curves algorithm, and refer to that instead ?

Comment (m-kuhn) [reply src/ui/qgsoptionsbase.ui:4908]: That depends on a plugin being enabled and it's hard to locate.
I would not mind putting a lot of text onto the tooltip (and copy paste it to the alg), I think tooltips can even have HTML, so a short abstract and a longer explanation could be separated into paragraphs?

Comment (olivierdalang) [reply src/ui/qgsoptionsbase.ui:4908]: Ok will try that

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**